ledger/narwhal/batch-header/src/lib.rs

```diff
@@ -56,6 +56,9 @@ pub struct BatchHeader<N: Network> {
 }
 
 impl<N: Network> BatchHeader<N> {
+    /// The maximum number of microcredits that can be spent on compute by the transactions in a batch.
+    /// This implies the block spend limit is bounded at `TRANSACTION_SPEND_LIMIT * N::NUM_MAX_CERTIFICATES`.
+    pub const BATCH_SPEND_LIMIT: u64 = N::TRANSACTION_SPEND_LIMIT * Self::MAX_TRANSMISSIONS_PER_BATCH as u64;
```

Comment:
Note: this would be identical to the current theoretical limit (and not an effective bound on block time) so we'll likely need this to be lower.

Reply:
If this is the limit we want to establish for now, there is no need for the migration logic above because there is fundamentally no change in possible bounds.

Optionally, for a more holistic approach, future change to this limit and migration can be introduced together with the future Fee reduction efforts. This would simplify the logic in the PR and the sister snarkOS PR to establish the logic behind enforcing the limits.